### PR TITLE
Add podspecs in correct Cocoapods layout

### PR DESCRIPTION
**Pull request by kuhnroyal**
Follow up to my comments in https://github.com/m0nac0/flutter-maplibre-gl/issues/91

I tried this with a private repo and it works well.
Alternatively you could create some `https://github.com/m0nac0/flutter-maplibre-specs` repo for this.

Build will fail until this is available in the default branch of either repository.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
@m0nac0 My tests with this Spec repo layout were successful. Everything in this PR works, just need a place to store the specs in the correct layout. Makes everything much easier.

You can test it locally with this repo: https://github.com/source-kitchen/flutter-maplibre-podspecs
You can clone this and use it as starting point for a new repo, don't fork as I plan to remove it ASAP.

**Comment by m0nac0** [comment]
@kuhnroyal I've created https://github.com/m0nac0/flutter-maplibre-podspecs. Would you like to adapt this PR?

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ dependencies:
     ...
     maplibre_gl:
       git:
-        url: https://github.com/m0nac0/flutter-maplibre-gl.git
+        url: https://github.com/m0nac0/flutter-maplibre-podspecs.git
```
Comment (m0nac0) [inline README.md]: This shouldn't be changed, because the Flutter plugin still lives in this repo

**Comment by m0nac0** [comment]
Not sure why the CI is failing

**Comment by kuhnroyal** [comment]
Yea me neither, I updated the example `Podfile.lock` and XCode files, hope that helps. The `Podfile.lock` should be committed to Git.

File: README.md (hunk 1 above)
Comment (kuhnroyal) [inline README.md]: I updated the wrong URL here...

**Comment by m0nac0** [comment]
Can you try re-adding the `pod update` step in the CI to check if that's the issue?

And when you tested this yourself, did you test with the example app or your own project?

**Comment by kuhnroyal** [comment]
It runs locally, so I am not sure, will try.

**Comment by kuhnroyal** [comment]
Not sure why, don't need that locally for the example or my own app.

**Comment by kuhnroyal** [comment]
The `pod update` was not needed, the problem was related to some Podfile miss configuration for the simulator target.

**Comment by kuhnroyal** [comment]
This should be fine now and can be squashed probably.

File: example/ios/Podfile [hunk 2]
```diff
@@ -39,13 +41,7 @@ target 'Runner' do
 end
```
Comment (m0nac0) [inline example/ios/Podfile]:
```suggestion
# commit 2c5aa27d7f039327306d1ecf80976778fd3ffcab from upstream#797 undone in PR #128
```

Can we please add this comment, that makes it easier for me to understand what we did here when there are e.g. any conflicts in the future when integrating new changes from upstream?

**Comment by m0nac0** [comment]
I added one small request, otherwise this looks good.

**Comment by kuhnroyal** [comment]
> I added one small request, otherwise this looks good.

Can't see it :)

**Comment by m0nac0** [comment]
Sorry, I left the comment as "pending" review and missed your reply. It should be visible now?

File: README.md (hunk 1 above)
Comment (kuhnroyal) [reply README.md]: Yea I think I reverted this

**Comment by kuhnroyal** [comment]
Done

**Review by m0nac0 — COMMENTED** [review]
One more thing: do we need example/pubspec.yaml to be committed to the repo or is this just an artifact of the change to .gitignore?

**Comment by kuhnroyal** [comment]
> One more thing: do we need example/pubspec.yaml to be committed to the repo or is this just an artifact of the change to .gitignore?

I would always commit the `pubspec.lock` for apps because this prevents accidental breakage, especially in CI.
For libraries this is not recommended but I believe for apps this is the correct way.

**Comment by m0nac0** [comment]
@kuhnroyal Thanks again for your patience!

**Pull request merged**